controllers/wordpressinstance_controller.go

```diff
@@ -71,31 +72,42 @@ func (r *WordpressInstanceReconciler) Reconcile(req ctrl.Request) (ctrl.Result,
 		}
 		return ctrl.Result{RequeueAfter: shortWait}, err
 	}
-
+	// Deletion is handled by Kubernetes garbage collector.
+	if meta.WasDeleted(wp) {
+		return ctrl.Result{Requeue: false}, nil
+	}
 	FillWithDefaults(wp)
 	if err := CreateMySQLInstance(ctx, r.Client, wp); err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+		wp.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
+		return ctrl.Result{RequeueAfter: shortWait}, r.Client.Status().Update(ctx, wp)
 	}
 	if err := CreateKubernetesCluster(ctx, r.Client, wp); err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+		wp.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
+		return ctrl.Result{RequeueAfter: shortWait}, r.Client.Status().Update(ctx, wp)
 	}
 	if err := CreateKubernetesApplication(ctx, r.Client, r.Scheme, wp); err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+		wp.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
+		return ctrl.Result{RequeueAfter: shortWait}, r.Client.Status().Update(ctx, wp)
 	}
 	if err := r.Client.Update(ctx, wp); err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+		wp.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
+		return ctrl.Result{RequeueAfter: shortWait}, r.Client.Status().Update(ctx, wp)
 	}
 
 	endpoint, err := GetEndpoint(ctx, r.Client, *wp)
 	if err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+		wp.Status.SetConditions(runtimev1alpha1.ReconcileError(err))
+		return ctrl.Result{RequeueAfter: shortWait}, r.Client.Status().Update(ctx, wp)
 	}
 	wp.Status.Endpoint = endpoint
-	if err := r.Client.Status().Update(ctx, wp); err != nil {
-		return ctrl.Result{RequeueAfter: shortWait}, err
+
+	wp.Status.SetConditions(runtimev1alpha1.Creating())
```

Comment:
when does this Creating status condition get cleared, does that happen when the Available status condition is set, or the ReconcileSuccess status condition is set?  or somehow else?

Reply:
There are 2 types of conditions: `Synced` and `Ready`. `Creating` and `Available` set the value of `Ready` type, while the ones starting with `Reconcile..` set the value for `Synced` type.
An example of how it looks:
```
status:
  conditions:
  - lastTransitionTime: "2019-12-30T15:25:56Z"
    reason: Managed resource is available for use
    status: "True"
    type: Ready
  - lastTransitionTime: "2019-12-30T15:25:56Z"
    reason: Successfully reconciled managed resource
    status: "True"
    type: Synced
```

When you set a condition for type `X`, the former value is overriden. So, it's always the most up-to-date one. You can look at here for the implementation https://github.com/crossplaneio/crossplane-runtime/blob/master/apis/core/v1alpha1/condition.go#L191

`reason`s of type `Ready` are worded for managed resources but this other PR fixes that since these conditions are already used by claims and other types. https://github.com/crossplaneio/crossplane-runtime/pull/99

Reply:
ah, I could have looked into that myself, sorry for being lazy @muvaf! 🙇

Reply:
No worries :)